controlpanel/api/aws.py

```diff
@@ -24,248 +23,418 @@ def iam_arn(resource, account=settings.AWS_ACCOUNT_ID):
-            PolicyName=policy_name,
+def create_group(name, path):
+    iam = boto3.resource('iam')
+    try:
+        iam.create_policy(
+            PolicyName=name,
             Path=path,
-            PolicyDocument=json.dumps(policy_document),
+            PolicyDocument=json.dumps(BASE_S3_ACCESS_POLICY),
         )
+    except iam.meta.client.exceptions.EntityAlreadyExistsException:
+        log.warning(f'Skipping creating policy {path}{name}: Already exists')
 
-    def create_policy_version(self, policy_arn, policy_document):
-        self._do(
-            "iam",
-            "create_policy_version",
-            PolicyArn=policy_arn,
-            PolicyDocument=json.dumps(policy_document),
-            SetAsDefault=True,
-        )
-        self._delete_policy_versions(policy_arn)
 
-    def get_policy(self, policy_arn):
-        return self._do("iam", "get_policy", PolicyArn=policy_arn)
+def update_group_members(group_arn, role_names):
+    policy = boto3.resource('iam').Policy(group_arn)
+    members = set(policy.attached_roles.all())
+    existing = {member.role_name for member in members}
 
-    def get_policy_version(self, policy_arn, version_id):
-        return self._do(
-            "iam", "get_policy_version", PolicyArn=policy_arn, VersionId=version_id
-        )
+    for role in role_names - existing:
+        policy.attach_role(RoleName=role)
 
-    def _detach_policy_from_roles(self, policy_arn):
-        """Detaches policy from all roles that it is attached to"""
+    for role in existing - role_names:
+        policy.detach_role(RoleName=role)
 
-        roles = self.list_entities_for_policy(policy_arn).get("PolicyRoles")
-        if roles:
-            for role in roles:
-                self.detach_policy_from_role(
-                    role_name=role["RoleName"], policy_arn=policy_arn
-                )
 
-    def delete_policy(self, policy_arn):
-        self._detach_policy_from_roles(policy_arn)
-        self._delete_policy_versions(policy_arn)
-        self._do("iam", "delete_policy", PolicyArn=policy_arn)
+def delete_group(group_arn):
+    policy = boto3.resource('iam').Policy(group_arn)
+    try:
+        policy.load()
+    except policy.meta.client.exceptions.NoSuchEntityException:
+        log.warning(f"Skipping deletion of policy {group_arn}: Does not exist")
+        return
 
-    def _delete_policy_version(self, policy_arn, version_id):
-        self._do(
-            "iam", "delete_policy_version", PolicyArn=policy_arn, VersionId=version_id
-        )
+    for role in policy.attached_roles.all():
+        policy.detach_role(RoleName=role.name)
 
-    def _delete_policy_versions(self, policy_arn):
-        versions = self.list_policy_versions(policy_arn).get("Versions", [])
-        for version in versions:
-            if not version["IsDefaultVersion"]:
-                self._delete_policy_version(policy_arn, version["VersionId"])
+    # for group in policy.attached_groups.all():
```

Comment:
Does this commented code need to be there?

Reply:
No, I'll remove it